src/ert/plugins/hook_implementations/forward_model_steps.py

```diff
@@ -264,6 +280,22 @@ def __init__(self) -> None:
             default_mapping={"<NUM_CPU>": 1, "<OPTS>": "", "<VERSION>": "version"},
         )
 
+    def validate_pre_experiment(self, fm_step_json: ForwardModelStepJSON) -> None:
+        if "<VERSION>" not in self.private_args:
+            raise ForwardModelStepValidationError(
+                "Forward model step ECLIPSE300 must be given a VERSION argument"
+            )
+        if shutil.which("eclrun") is not None:
+            pm = ErtPluginManager()
+            ecl_env = _ecl_env_path(pm.get_ecl300_config_path())
+            version = self.private_args["<VERSION>"]
+            available_versions = _available_eclrun_versions(ecl_env, "e300")
+            if version not in available_versions:
+                raise ForwardModelStepValidationError(
+                    f"Unavailable ECLIPSE300 version {version} current supported "
+                    f"versions {available_versions}"
+                )
```

Comment:
Could this be one function taking in ecl_env and two strings, "e300" and "ECLIPSE300" ?
In the previous case it would be ecl_env, "eclipse" and "ECLIPSE100" ?

Reply:
After some discussion with @yngve-sk and some iterations, we decided to move all the logic inside one function `_available_eclrun_versions` that will only take the simulator name() as an argument.